src/TaylorBot.Net/Core/src/TaylorBot.Net.Commands/Parsers/Users/ParsedUser.cs

```diff
@@ -28,7 +28,10 @@ public UserParser(ITaylorBotClient taylorBotClient, IUserTracker userTracker)
                 return Error(new ParsingFailed("User option is required."));
             }
 
-            var user = await _taylorBotClient.ResolveRequiredUserAsync(new(optionValue.Value.GetString()!));
+            var user = context.Guild != null
+                ? await _taylorBotClient.ResolveGuildUserAsync(context.Guild, new(optionValue.Value.GetString()!)) ??
```

Comment:
Nit: can we move the the id into a variable just so it makes this code a little less verbose?
```cs
SnowflakeId userId = new(optionValue.Value.GetString()!);
```